src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -874,6 +881,19 @@ private TypedConstant GetDefaultValueArgument(ParameterSymbol parameter, Attribu
             {
                 return new TypedConstant(parameterType, kind, defaultValue);
             }
+
+            static int getArgumentIndex(ParameterSymbol parameter, ImmutableArray<string> constructorArgumentNamesOpt)
+            {
+                if (constructorArgumentNamesOpt.IsDefault || parameter.CallerArgumentExpressionParameterIndex == -1)
+                {
+                    return parameter.CallerArgumentExpressionParameterIndex;
+                }
+
+                Debug.Assert(parameter.ContainingSymbol is MethodSymbol);
+                var methodSymbol = (MethodSymbol)parameter.ContainingSymbol;
+                var callerArgumentParameter = methodSymbol.Parameters[parameter.CallerArgumentExpressionParameterIndex];
+                return constructorArgumentNamesOpt.IndexOf(callerArgumentParameter.Name);
```

Comment:
>return constructorArgumentNamesOpt.IndexOf(callerArgumentParameter.Name); [](http://example.com/codeflow?start=16&length=73)

It looks like this code path isn't tested. Especially it is interesting what the behavior is for the situation when the corresponding argument isn't actually named, but others are.

Reply:
Also when there is an out of order named argument used at the position of the target parameter.